lib/gc/Transforms/VerifyTargetDescription.cpp

```diff
@@ -0,0 +1,115 @@
+//===-- VerifyTargetDescription.cpp - Verity target desc --------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "gc/Analysis/TargetDescriptionAnalysis.h"
+#include "gc/Transforms/Passes.h"
+#include "mlir/Dialect/DLTI/DLTI.h"
+#include "mlir/IR/BuiltinOps.h"
+
+#include "mlir/Pass/Pass.h"
+using namespace mlir;
+
+namespace mlir {
+namespace gc {
+#define GEN_PASS_DEF_VERIFYTARGETDESCRIPTION
+#include "gc/Transforms/Passes.h.inc"
+
+namespace {
+
+static LogicalResult verifyCPUTargetDescription(RewriterBase &rewriter,
+                                                Operation *op) {
+  CPUTargetDescriptionAnalysis cpuTargetDesc(op);
+  Location loc = op->getLoc();
+
+  // Check if the num_threads is existed and greater than 0
+  std::optional<Attribute> numThreadsAttr =
+      cpuTargetDesc.getPropertyValue(CPUTargetDescriptionAnalysis::kNumThreads);
+  if (numThreadsAttr) {
+    if (!isa<IntegerAttr>(*numThreadsAttr) ||
+        cpuTargetDesc.getNumThreads() < 1) {
+      mlir::emitError(loc)
+          << "num_threads must be a greater than 0 integer, but get "
+          << *numThreadsAttr;
+      return failure();
+    }
+  }
+
+  // Check if the L1 cache size is existed and greater than 0
+  std::optional<Attribute> l1CacheSizeAttr = cpuTargetDesc.getPropertyValue(
+      CPUTargetDescriptionAnalysis::kL1CacheSize);
+  if (l1CacheSizeAttr) {
+    if (!isa<IntegerAttr>(*l1CacheSizeAttr) ||
+        cpuTargetDesc.getCacheSize(1) < 1) {
+      mlir::emitError(loc)
```

Comment:
It looks like we have supported `StringAttr` as input in `getIntFromAttribute`, shall we cover it as well?

Reply:
Yes, the test has been added here https://github.com/intel/graph-compiler/blob/1731525ae48e9b2b3624145c687bffc8db03fd8f/test/mlir/unittests/Analysis/TargetDescriptionAnalysisTest.cpp#L32

Reply:
The check has been removed here